controllers/object_controls.go

```diff
@@ -1344,15 +1350,44 @@ func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec,
 		configEnvvarName = "CRIO_CONFIG"
 	}
 
-	setContainerEnv(mainContainer, "RUNTIME_CONFIG", DefaultRuntimeConfigTargetDir+sourceConfigFileName)
-	setContainerEnv(mainContainer, configEnvvarName, DefaultRuntimeConfigTargetDir+sourceConfigFileName)
+	// Handle the top-level configs
+	if runtimeConfigFiles.topLevelConfigFile != "" {
+		sourceConfigFileName := path.Base(runtimeConfigFiles.topLevelConfigFile)
+		sourceConfigDir := path.Dir(runtimeConfigFiles.topLevelConfigFile)
+		containerConfigDir := DefaultRuntimeConfigTargetDir
+		setContainerEnv(mainContainer, "RUNTIME_CONFIG", containerConfigDir+sourceConfigFileName)
+		setContainerEnv(mainContainer, configEnvvarName, containerConfigDir+sourceConfigFileName)
+
+		volMountConfigName := fmt.Sprintf("%s-config", runtime)
+		volMountConfig := corev1.VolumeMount{Name: volMountConfigName, MountPath: containerConfigDir}
+		mainContainer.VolumeMounts = append(mainContainer.VolumeMounts, volMountConfig)
+
+		configVol := corev1.Volume{Name: volMountConfigName, VolumeSource: corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{Path: sourceConfigDir, Type: newHostPathType(corev1.HostPathDirectoryOrCreate)}}}
+		obj.Spec.Template.Spec.Volumes = append(obj.Spec.Template.Spec.Volumes, configVol)
+	}
+
+	// Handle the drop-in configs
+	// TODO: It's a bit of a hack to skip the `nvidia-kata-manager` container here.
+	// Ideally if the two projects are using the SAME API then this should be
+	// captured more rigorously.
+	// Note that we probably want to implement drop-in file support in the
+	// kata manager in any case -- in which case it will be good to use a
+	// similar implementation.
+	if runtimeConfigFiles.dropInConfigFile != "" && containerName != "nvidia-kata-manager" {
+		sourceConfigFileName := path.Base(runtimeConfigFiles.dropInConfigFile)
+		sourceConfigDir := path.Dir(runtimeConfigFiles.dropInConfigFile)
+		// TODO: This should be a constant.
+		containerConfigDir := "/runtime/config-dir.d/"
+		setContainerEnv(mainContainer, "RUNTIME_DROP_IN_CONFIG", containerConfigDir+sourceConfigFileName)
+		setContainerEnv(mainContainer, "RUNTIME_DROP_IN_CONFIG_HOST_PATH", runtimeConfigFiles.dropInConfigFile)
```

Comment:
Is `RUNTIME_DROP_IN_CONFIG_HOST_PATH` an actual envvar that the toolkit container processes? I did a quick search and can't find any mention of this variable. As a consequence, when testing out this PR I see that the toolkit container is setting 
```
imports = ["/runtime/config-dir.d/*.toml"]
```
in the top-level containerd config instead of 
```
imports = ["//run/toolkit/config/*.toml"]
```